src/adapter/src/catalog/storage.rs

```diff
@@ -1000,19 +1001,21 @@ impl<'a, S: Append> Transaction<'a, S> {
                 Some(schema) => schema,
                 None => return,
             };
-            let database_id = match schema.database_id {
-                Some(id) => id,
-                None => return,
-            };
-            let _database = match databases.get(&DatabaseKey { id: database_id }) {
-                Some(database) => database,
-                None => return,
+            let database_spec = match schema.database_id {
+                Some(id) => {
+                    if databases.get(&DatabaseKey { id }).is_none() {
+                        error!("unknown database id {id}");
```

Comment:
Under what circumstances can this error occur?

Reply:
If I had to guess, if this error is hit or the return on line 1002 are hit, then something really bad must have happened.

I just wasn't sure if they were included for a specific reason and therefore didn't want to change the existing behavior because I didn't really understand it.

Should I just turn them both into panics?

Reply:
For example I think this would happen if a database was deleted but an object in the database wasn't deleted, in which the stash is likely corrupted somehow.

Reply:
I'm in favor of panicing if this is the case.

Reply:
Done